assets/app/scripts/controllers/overview.js

```diff
@@ -40,22 +53,156 @@ angular.module('openshiftConsole')
-        var ls = new LabelSelector(service.spec.selector);
-        var servicePods = ls.select($scope.pods)
-        $scope.podsByServiceByLabel[name]  =  {};
-        DataService.objectsByAttribute(servicePods, "metadata.labels", $scope.podsByServiceByLabel[name], null, "metadata.name");
+    // Expects deploymentsByServiceByDeploymentConfig to be up to date
+    var podRelationships = function() {
+      $scope.monopodsByService = {"": {}};
+      $scope.podsByService = {};
+      $scope.podsByDeployment = {};
+
+      // Initialize all the label selectors upfront
+      var depSelectors = {};
+      angular.forEach($scope.deployments, function(deployment, depName){
+        depSelectors[depName] = new LabelSelector(deployment.spec.selector);
+        $scope.podsByDeployment[depName] = {};
+      });
+      var svcSelectors = {};
+      angular.forEach($scope.unfilteredServices, function(service, svcName){
+        svcSelectors[svcName] = new LabelSelector(service.spec.selector);
+        $scope.podsByService[svcName] = {};
+      });      
+
+
+      angular.forEach($scope.pods, function(pod, name){
+        var deployments = [];
+        var services = [];
+        angular.forEach($scope.deployments, function(deployment, depName){
+          var ls = depSelectors[depName];
+          if (ls.matches(pod)) {            
+            deployments.push(depName);
+            $scope.podsByDeployment[depName][name] = pod;
+          }
+        });
+        angular.forEach($scope.unfilteredServices, function(service, svcName){
+          var ls = svcSelectors[svcName];
+          if (ls.matches(pod)) {
+            services.push(svcName);
+            $scope.podsByService[svcName][name] = pod;
+
+            var isInDepInSvc = false;
+            // Not using angular.forEach so that we can break out early
+            for (var i = 0; !isInDepInSvc && i < deployments.length; i++) {
+              var depName = deployments[i];
+              if ($scope.deploymentsByService[svcName][depName]) {
+                isInDepInSvc = true;
+              }
+            }
+
+            if (!isInDepInSvc) {
+              $scope.monopodsByService[svcName] = $scope.monopodsByService[svcName] || {};
+              $scope.monopodsByService[svcName][name] = pod;
+            }
+          }
+        });
+        if (deployments.length == 0 && services.length == 0 && showMonopod(pod)) {
+          $scope.monopodsByService[""][name] = pod;
+        }
+      });
+
+      console.log("podsByDeployment", $scope.podsByDeployment);      
+      console.log("podsByService", $scope.podsByService); 
+      console.log("monopodsByService", $scope.monopodsByService); 
+    };
+
+    // Filter out monopods we know we don't want to see
+    var showMonopod = function(pod) {
+      // Hide pods in the Succeeded or Terminated phase since these are run once pods
+      // that are done
+      if (pod.status.phase == 'Succeeded' || pod.status.phase == 'Terminated') {
```

Comment:
Add a TODO... we may want to show pods for some period of time after they complete, just like recent builds